File: mobile-client/android/UseCase/Catalog/Browse/ListCatalogNodesUseCase.kt
package app.mobiling.client.client.usecase.catalog.browse

import app.mobiling.client.client.contract.catalog.browse.CatalogNodeSummary
import app.mobiling.client.client.contract.catalog.browse.ListCatalogNodesQuery
import app.mobiling.client.client.data.catalog.browse.CatalogBrowseGateway

/**
 * Marketing America Corp. Oleksandr Tishchenko
 */
class ListCatalogNodesUseCase(
    private val gateway: CatalogBrowseGateway,
) {
    suspend operator fun invoke(query: ListCatalogNodesQuery): List<CatalogNodeSummary> = gateway.listNodes(query)
}

File: client/ios/UseCase/Catalog/Browse/ListCatalogNodesUseCase.swift
import Foundation

// Marketing America Corp. Oleksandr Tishchenko
public struct ListCatalogNodesUseCase {
    public let gateway: CatalogBrowseGateway

    public init(gateway: CatalogBrowseGateway) {
        self.gateway = gateway
    }

    public func callAsFunction(query: ListCatalogNodesQuery) async throws -> [CatalogNodeSummary] {
        try await gateway.listNodes(query: query)
    }
}
